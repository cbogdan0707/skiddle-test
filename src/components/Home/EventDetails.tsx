import React, { ReactElement } from 'react';
import { Link } from 'react-router-dom';

interface EventProps {
  imageUrl: string;
  eventName: string;
  description: string;
  location: string;
  date: string;
  id: number;
}

export const EventDetails = ({
  eventDetails,
}: {
  eventDetails: EventProps;
}): ReactElement => {
  return (
    <div className="skiddle-c-event-details">
      <img alt="event" src={eventDetails.imageUrl} />
      <div className="skiddle-u-text-align-center skiddle-u-bold skiddle-u-mt-small">
        {eventDetails.eventName}
      </div>
      <div className="skiddle-u-mt-small">{eventDetails.description}</div>
      <div className="skiddle-u-mt-small skiddle-c-location">
        {eventDetails.location}
      </div>
      <div className="skiddle-u-mt-small skiddle-c-date">
        {eventDetails.date}
      </div>
      <div className="skiddle-u-text-align-center skiddle-u-mt-medium">
        <Link
          className="skiddle-u-primary-button skiddle-u-pa-small"
          to={{ pathname: `/event/${eventDetails.id}` }}
        >
          view details
        </Link>
      </div>
    </div>
  );
};
