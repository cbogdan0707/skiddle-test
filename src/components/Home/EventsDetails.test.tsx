import React from 'react';
import { render } from '@testing-library/react';
import { EventDetails } from './EventDetails';
import { BrowserRouter } from 'react-router-dom';

const eventDetails = {
  imageUrl: 'test/imageUrl',
  eventName: 'test name',
  description: 'test description',
  location: 'test location',
  date: 'test date',
  id: 232345,
};

describe('Event details', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <EventDetails eventDetails={eventDetails} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
