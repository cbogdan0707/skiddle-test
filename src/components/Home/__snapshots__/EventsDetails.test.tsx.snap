// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event details should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="skiddle-c-event-details"
  >
    <img
      alt="event"
      src="test/imageUrl"
    />
    <div
      class="skiddle-u-text-align-center skiddle-u-bold skiddle-u-mt-small"
    >
      test name
    </div>
    <div
      class="skiddle-u-mt-small"
    >
      test description
    </div>
    <div
      class="skiddle-u-mt-small skiddle-c-location"
    >
      test location
    </div>
    <div
      class="skiddle-u-mt-small skiddle-c-date"
    >
      test date
    </div>
    <div
      class="skiddle-u-text-align-center skiddle-u-mt-medium"
    >
      <a
        class="skiddle-u-primary-button skiddle-u-pa-small"
        href="/event/232345"
      >
        view details
      </a>
    </div>
  </div>
</DocumentFragment>
`;
